js/app/modules/contentGroup/contentGroup.js

```diff
@@ -217,6 +227,30 @@ const ContentGroup = new Module.Class({
         if (!this._selection.in_error_state)
             return;
         this._stack.visible_child_name = ERROR_PAGE_NAME;
+        this._log_button_stack.visible_child_name = 'button';
+    },
+
+    _on_log_button_click: function () {
+        let timestamp = new Date(Date.now());
+        let stamp = 'eos-knowledge-lib log ' + timestamp.toISOString() + '.txt';
+        let log_name = GLib.build_filenamev([GLib.get_home_dir(), stamp]);
```

Comment:
b) I was thinking it would be better to have them out in the open. For one thing, there will only be a log file if the user clicks the button, so presumably they will only click it once. For another thing, we want the user to be able to find it easily when they report the issue to customer support.